src/mesh.c

```diff
@@ -0,0 +1,365 @@
+
+static int mesh_read_dropped_frames_congestion(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " DROPPED_FRAMES_CONGESTION
+                    " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, DROPPED_FRAMES_CONGESTION, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " DROPPED_FRAMES_CONGESTION);
+    return ret;
+  }
+
+  return mesh_submit(m, DROPPED_FRAMES_CONGESTION, value);
+} /* int mesh_read_dropped_frames_congestion */
+
+static int mesh_read_dropped_frames_no_route(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " DROPPED_FRAMES_NO_ROUTE
+                    " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, DROPPED_FRAMES_NO_ROUTE, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " DROPPED_FRAMES_NO_ROUTE);
+    return ret;
+  }
+
+  return mesh_submit(m, DROPPED_FRAMES_NO_ROUTE, value);
+} /* int mesh_read_dropped_frames_no_route */
+
+static int mesh_read_dropped_frames_ttl(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " DROPPED_FRAMES_TTL " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, DROPPED_FRAMES_TTL, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " DROPPED_FRAMES_TTL);
+    return ret;
+  }
+
+  return mesh_submit(m, DROPPED_FRAMES_TTL, value);
+} /* int mesh_read_dropped_frames_ttl */
+
+static int mesh_read_fwded_frames(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " FWDED_FRAMES " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, FWDED_FRAMES, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " FWDED_FRAMES);
+    return ret;
+  }
+
+  return mesh_submit(m, FWDED_FRAMES, value);
+} /* int mesh_read_fwded_frames */
+
+static int mesh_read_fwded_unicast(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " FWDED_UNICAST " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, FWDED_UNICAST, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " FWDED_UNICAST);
+    return ret;
+  }
+
+  return mesh_submit(m, FWDED_UNICAST, value);
+} /* int mesh_read_fwded_unicast */
+
+static int mesh_read_fwded_mcast(const mesh_list_t *m) {
+  int ret;
+  value_t value;
+
+  DEBUG(PLUGIN_NAME ": Reading attribute " FWDED_MCAST " for device %s",
+        m->net_name);
+
+  ret = mesh_read_stats_attr(m, FWDED_MCAST, &value.derive);
+
+  if (ret != 0) {
+    ERROR(PLUGIN_NAME " : Unable to read " FWDED_MCAST);
+    return ret;
+  }
+
+  return mesh_submit(m, FWDED_MCAST, value);
+} /* int mesh_read_fwded_mcast */
+
+typedef int (*mesh_reader_t)(const mesh_list_t *);
+
+static const mesh_reader_t g_mesh_readers[] = {
```

Comment:
This array is only used inside `mesh_read()`, so you can just as well move it there.